setuptools_rust/build.py

```diff
@@ -207,7 +207,7 @@ def build_extension(
 
         # Execute cargo
         try:
-            stderr = subprocess.PIPE if quiet else None
+            stderr = None if quiet else subprocess.PIPE
```

Comment:
I'm pretty sure the original condition was what I intended. My understanding was:

- if `quiet` we want to capture all output from cargo, so we use `subprocess.PIPE`
- otherwise, pass `None` so that `cargo` will inherit the stderr handle from this process

Reply:
I think there might be some confusion w.r.t. the `if not quiet:` clause further below. By using `stderr=subprocess.PIPE` with `check_output`, the stderr will essentially be discarded as `check_output` explicitly returns only `stdout`. Maybe `stderr=subprocess.STDOUT` to merge stderr into stdout and print both in line 221 would be more helpful?

Reply:
> * if `quiet` we want to capture all output from cargo, so we use `subprocess.PIPE`

Do you mean capture-and-discard the output from `cargo`? Because capture-and-show would be the opposite of quiet.

I admit that I don't understand the setup of `check_output` well, but the way I read the [docs](https://docs.python.org/3/library/subprocess.html#subprocess.check_output) is that the `stderr=PIPE` is actually necessary to capture the output in the first place (whether it is then discarded or shown), in particular: 

> This is equivalent to:
>
> ```run(..., check=True, stdout=PIPE).stdout```

and `subprocess.run` not capturing stderr by default.

Reply:
When I tried subprocess.STDOUT in #254, it [corrupted](https://github.com/PyO3/setuptools-rust/pull/254#discussion_r908114339) the JSON result of `cargo metadata`.

PS. Sorry, your first comment hadn't shown up in my open tab before I responded.

Reply:
@davidhewitt @adamreichold 
Any ideas on how to move forward on this? I think we can agree that how stdin/stout are handled are somewhat confusing. Perhaps it might make sense to not use the `check_output` wrapper, but rather `run` directly, so we have more explicit control (or at least clearer documentation)?

Reply:
@h-vetinari what do you think of #268?

Reply:
Fine by me, thanks 👍 

Would it make sense to add a test that switches on `CARGO_LOG=trace` and expects a minimum number of lines of output?

Reply:
Sure thing, added in https://github.com/PyO3/setuptools-rust/pull/268/commits/6eb0ab89fa561eb21097f4cd71e038de2e1635cb